src/main/webapp/2Dscript.js

```diff
@@ -80,7 +169,361 @@ function addCenterControl(controlDiv) {
+            // console.log(subLine2D); Can leave in extra parameters and that
+            // has no effect on the graphing.
+            subPath = new google.maps.Polyline({
+              path: subLine,
+              geodesic: true,
+              strokeColor: 'blue',
+              strokeOpacity: 1.0,
+              strokeWeight: 2,
+            });
+            // Setup event listener to show option for 3D window when polyline
+            // is clicked.
+            google.maps.event.addListener(subPath, 'click', function(event) {
+              const latLng = event.latLng;
+              console.log('Polyline clicked at lat: ' + latLng.lat() +
+                ' lng: ' + latLng.lng());
+              infoWindow = new google.maps.InfoWindow({
+                content: '<a href=/home.html?id=' + id + '&dataType=' +
+                  type + '&subSection=' + subSectionNumber +
+                  '&stored=true' + '> View in 3D </a>',
+                position: latLng,
+              });
+              infoWindow.setMap(map);
+            },
+            );
+
+            subPath.setMap(map);
+
+            markerBottom = new google.maps.Marker({
+              position: {lat: discoveredMinLat, lng: discoveredMinLatPair},
+              title: 'Lat: ' + discoveredMinLat +
+                ' Lng: ' + discoveredMinLatPair,
+            });
+            markerTop = new google.maps.Marker({
+              position: {lat: discoveredMaxLngPair, lng: discoveredMaxLng},
+              title: 'Lat: ' + discoveredMaxLngPair +
+                ' Lng: ' + discoveredMaxLng,
+            });
+            // At this point iterate through all the values to contruct a new
+            // line with the appropriate markers.
+            markerBottom.setMap(map);
+            markerTop.setMap(map);
+            // Constant to account for possibility of multiple subsections.
+
+            sessionStorage.setItem(id + '_' + type + '_' +
+              subSectionNumber, JSON.stringify(subLine));
+          }
+        }
+      }
+    },
+    mousemove: function(event) {
+      if (ismousedown === true) {
+        console.log('Set width triggered');
+        finX = event.pageX;
+        finY = event.pageY;
+        widget.width(finX - x);
+        widget.height(finY - y);
+        widget.css({
+          'width': (finX - x) + '!important',
+          'height': (finY - y) + '!important',
+          'display': 'block',
+          'border': '2px dashed #ccc',
+        });
+      }
+    },
+  });
+});
+
+/**
+ * Check whether the given lat, lng is within the bounding rectangle.
+ * @param {number} minLat
+ * @param {number} maxLat
+ * @param {number} minLng
+ * @param {number} maxLng
+ * @param {number} valLat
+ * @param {number} valLng
+ * @return {boolean}
+ */
+function withinBound(minLat, maxLat, minLng, maxLng, valLat, valLng) {
+  if ((valLat >= minLat && valLat <= maxLat) &&
+    (valLng >= minLng && valLng <= maxLng)) {
+    return true;
+  }
+  return false;
```

Comment:
This can be simplified:
`return lat >= minLat && lat <= maxLat && lng >= minLng && lng <= maxLng;`